web-components/src/components/robotoff/robotoff-question.ts

```diff
@@ -0,0 +1,161 @@
+import { LitElement, html, css, nothing } from "lit"
+import { customElement, property } from "lit/decorators.js"
+import {
+  currentQuestionIndex,
+  fetchQuestionsByProductCode,
+  nextQuestion,
+  isQuestionsFinished,
+  questions,
+  hasQuestions,
+  numberOfQuestions,
+  hasAnswered,
+} from "../../signals/questions"
+import { Task } from "@lit/task"
+import { localized, msg } from "@lit/localize"
+import { EventType, IS_DEVELOPMENT_MODE } from "../../constants"
+import { QuestionStateEventDetail } from "../../types"
+
+/**
+ * An example element.
+ *
+ * @fires count-changed - Indicates when the count changes
+ * @slot - This element has a slot
+ * @csspart button - The button
+ */
+@customElement("robotoff-question")
+@localized()
+export class RobotoffQuestion extends LitElement {
+  static override styles = css`
+    :host {
+      display: block;
+    }
+    .question-wrapper {
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      text-align: center;
+    }
+
+    .message {
+      font-style: italic;
+    }
+  `
+  /**
+   * Options for the component
+   * @type {Object}
+   * @property {boolean} showMessage - Whether to show the message
+   * @property {boolean} showLoading - Whether to show the loading indicator
+   * @property {boolean} showError - Whether to show the error message
+   */
+  @property({ type: Object, reflect: true })
+  options: {
+    showMessage?: boolean
+    showLoading?: boolean
+    showError?: boolean
+  } = {}
+
+  /**
+   * The product id to fetch questions for
+   * @type {string}
+   */
+  @property({ type: String, attribute: "product-id" })
+  productId: string = ""
+
+  /**
+   * The insight types to filter questions separate by comma
+   * @type {string}
+   */
+  @property({ type: String, attribute: "insight-types" })
+  insightTypes: string = ""
+
+  private _questionsTask = new Task(this, {
+    task: async ([productId, insightTypes], {}) => {
+      if (!productId) {
+        return []
+      }
+      const params = insightTypes ? { insight_types: insightTypes } : {}
+      await fetchQuestionsByProductCode(productId, params)
+      this._emitQuestionStateEvent()
+      return questions.get()
+    },
+    args: () => [this.productId, this.insightTypes],
+  })
+
+  private _emitQuestionStateEvent = () => {
+    const detail: QuestionStateEventDetail = {
+      index: currentQuestionIndex.get(),
+      numberOfQuestions: numberOfQuestions.get(),
+    }
+    this.dispatchEvent(
+      new CustomEvent(EventType.QUESTION_STATE, {
+        detail,
+        bubbles: true,
+        composed: true,
+      })
+    )
```

Comment:
I was proposing to use https://www.npmjs.com/package/@lit-app/state instead of messing with signals. Do you think it's overkill ?